xbmc/utils/HevcSei.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "HevcSei.h"
+
+void hevc_add_start_code_emulation_prevention_3_byte(std::vector<uint8_t>& buf)
+{
+  auto count = buf.size();
+  size_t i = 0;
+
+  while (i < count)
+  {
+    if (i > 2 && buf[i - 2] == 0 && buf[i - 1] == 0 && buf[i] <= 3)
+    {
+      buf.insert(buf.begin() + i, 3);
+      count += 1;
```

Comment:
`count` variable seems redundant as is increased at same time as `buf.insert`

Use directly `while (i < buf.size())` should be equivalent.

Reply:
Yes it is the same. I wasn't sure if `buf.size()` is as performant.